rift/engine.py

```diff
@@ -139,13 +139,12 @@ def default_physical_interface(self):
         sys.exit(1)
 
     def nr_nodes(self):
-        total_nr_nodes = 0
         if 'shards' in self._config:
             for shard_config in self._config['shards']:
                 if 'nodes' in shard_config:
-                    for _node_config in shard_config['nodes']:
-                        total_nr_nodes += 1
-        return total_nr_nodes
+                    return len(shard_config['nodes'])
```

Comment:
@brunorijsman I've simplified this method, removing a for loop.

Reply:
The new code returns the wrong value if there is more than one shard. It returns the number of nodes in the first shard, not the total number of nodes across all shards. I suggest going back to the original code.

Some additional notes. My code currently does not support the concept of shards; I always run all shards in the same Python process. In the early implementation I allowed the shard syntax to be used to be compatible with the config file format that Tony uses in the Juniper code (in later code I added some extensions to my config file syntax that Tony does not support, but AFAIK my syntax is still a superset of Tony's)